src/resources/assets/js/annotations/mixins/ContextLayerMixin.vue

```diff
@@ -0,0 +1,42 @@
+<script>
+import TileLayer from 'ol/layer/Tile';
+import TileWMS from 'ol/source/TileWMS';
+
+
+
+/**
+ * Mixin for the annotationCanvas component that contains logic for the Context-Layer (Geo-Overlay) interaction.
+ *
+ * @type {Object}
+ */
+export default {
+    data() {
+        return {
+            overlays: [],
+        };
+    },
+    methods: {
+
+    },
+    mounted() {
+        this.overlays = biigle.$require('annotations.overlays');
+        // console.log('geoOverlays: ', this.overlays);
+        // console.log('activeId: ', this.settings);
+
```

Comment:
@mzur,
I am a bit confused by all the  imports between the biigle modules, which is why I pose this question:

In the geo-module I created a `ContextLayerMixin` for the `annotationCanvas.vue` of biigle/core. 
In this mixin, I want to access a couple of variables from the annotation settings tab (i.e. the [activeId](https://github.com/biigle/geo/pull/60/files#diff-94f4a070c8c000dbc91671c354a3b1906519c22d3b803e9917730f9b3c4c3e4eR56) or [opacity](https://github.com/biigle/geo/pull/60/files#diff-94f4a070c8c000dbc91671c354a3b1906519c22d3b803e9917730f9b3c4c3e4eR69) of the geo-overlay settings, which are part of yet another plugin that is published from the geo-module to the `settingsTab.vue` in core).
In order to access these variables, do I have to make them available as properties in the `annotationCanvas` component ([here](https://github.com/biigle/core/blob/5f6861a69eef53fa1e22b77be8e0d2c1e2231ccf/resources/views/annotations/show.blade.php#L66)) in biigle/core, or can I directly import them, for example the whole settings-variable from the [settings-store](https://github.com/biigle/core/blob/5f6861a69eef53fa1e22b77be8e0d2c1e2231ccf/resources/assets/js/annotations/stores/settings.js#L4)?

The first option should work because the variables would be made accessible within the `ContextLayerMixin`, but it would require changes to the code in biigle/core, which doesn't seem right. Regarding the second option, I'm not sure how to make the settings-store variable available.

Reply:
I see your point, there is no direct way to access the settings in an annotation canvas mixin. Maybe you can do it like the reports module. It does everything in the [settings tab plugin](https://github.com/biigle/reports/blob/master/src/resources/assets/js/annotations/components/settingsTabPlugin.vue) by using the `annotations.map.init` and `images.change` events to hook up OpenLayers objects and update image data.